### PR TITLE
Add z-index to sticky submit button

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

Fixes [USH-4564](https://dimagi.atlassian.net/browse/USH-4564), and puts the anchored submit button in front of a form's datetime field.

Before:

Screenshot 2024-05-23 at 10 07 50 AM

After:

Screenshot 2024-05-23 at 11 51 54 AM

## Technical Summary

The datetime field uses the Bootstrap class `form-control`, which adds a z-index of 2 (3 if focused). So this PR adds a z-index=4 to the sticky submit button.

Hoping I went about updating the different files related to the Bootstrap migration correctly, please feel free to correct me around that.

## Feature Flag

WEB_APPS_ANCHORED_SUBMIT

### Safety story

- Tested locally
- Only updated the z-index of a single element. It's hard to see any risks of the sticky submit button being at a layer closer to the front.

### QA Plan

QA does weekly testing of web apps UI. This should go through that before merged/deployed

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-4564]: https://dimagi.atlassian.net/browse/USH-4564?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/form.scss [hunk 1]
```diff
@@ -33,6 +33,9 @@ $group-indent: 15px;
 
   .sticky-submit {
     position: sticky;
+    /* Puts the anchored submit button in front of form elements that use bootstrap's .form-control,
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/form.scss:36]: Is sticky submit only implemented on web apps, not app preview?

**Review by MartinRiese — APPROVED** [review]

**Review by orangejenny — APPROVED** [review]
Looks reasonable. Other things to consider that sometimes trip up z-index - do these appear above or below the sticky submit bar? This code shouldn't make any of these worse, just wondering if there are other items that appear on top of the bar.
* Question widgets, like the calendar display for date question
* Modals
* Dropdowns - are there any that would collide with the submit bar? Probably not...the language dropdown probably won't ever be long enough to reach the bottom of the screen.

File: corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/form.scss (hunk 1 above)
Comment (MartinRiese) [reply corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/form.scss:36]: If I remember correctly there was some logic to exclude it from the preview. https://github.com/dimagi/commcare-hq/pull/34490/files#diff-a0495e536c5711042a5da5dfbcd9c8976f0c009df8e52222479ab2d7f488ed89R50

**Review by Jtang-1 — APPROVED** [review]

**Comment by orangejenny** [comment]
@AddisonDunn Thank you for checking, I agree that's fine. Was that testing locally or on staging?

**Comment by orangejenny** [comment]
@AddisonDunn Yeah, the difference is probably related to the bootstrap version. Thanks for checking, I'll add it to the B5 bug list.

**Comment by orangejenny** [comment]
@AddisonDunn After interacting with a form locally, I think the B5 behavior is okay. The field disappears behind the submit bar once it loses focus. It's weird either way - it wouldn't be ideal for the field to be hidden while someone is working in it. I'm going to leave this alone.

**Pull request merged**